src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -4897,7 +4897,7 @@ void MergeTreeData::removeQueryId(const String & query_id) const
 {
     std::lock_guard lock(query_id_set_mutex);
     if (query_id_set.find(query_id) == query_id_set.end())
-        LOG_WARNING(log, "We have query_id removed but it's not recorded. This is a bug");
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "We have query_id removed but it's not recorded. This is a bug");
```

Comment:
The method is used in dtor. Shall we still throw?

Reply:
It's ok to abort on logical error in some cases. But we can replace `throw ...` with `assert(false)` to abort in debug builds only.

Reply:
Yeah, that would be consistent with other code.